dbms/src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -482,12 +471,12 @@ void getArrayJoinedColumns(ASTPtr & query, SyntaxAnalyzerResult & result, const
             else /// This is a nested table.
             {
                 bool found = false;
-                for (const auto & column_name : source_columns)
+                for (const auto & column : source_columns)
```

Comment:
It's strange but if we change ordered ```source_columns``` to unordered ```source_columns_set``` with the same data we get an error with names resolving.